File: 2129-number-of-pairs-of-interchangeable-rectangles/number-of-pairs-of-interchangeable-rectangles.cpp
class Solution {
public:
    long long interchangeableRectangles(vector<vector<int>>& rectangles) {
        map<double,int>mpp;
        for(int i=0;i<rectangles.size();i++){
            int t=rectangles[i][0];
            int y=rectangles[i][1];
            mpp[(1.0* t)/y]++;
        }
        long long  count=0;
        for(auto & p :mpp){
            int t=p.second;
            count+=(long long )t*(long long )(t-1)/2;
        }
        return count;
    }
};

File: 2129-number-of-pairs-of-interchangeable-rectangles/number-of-pairs-of-interchangeable-rectangles.java
class Solution {
    public long interchangeableRectangles(int[][] rectangles) {
        Map<Double,Integer>mpp=new HashMap<>();
        for(int i=0;i<rectangles.length;i++){
            int first=rectangles[i][0];
            int second=rectangles[i][1];
            double t=1.0 *first/second;
            if(mpp.containsKey(t)){
                mpp.put(t,mpp.get(t)+1);
            }
            else{
                mpp.put(t,1);
            }
        }
        long count=0;
        for(Double key : mpp.keySet()){
            int val=mpp.get(key);
            count+=(long)(val)*(long)(val-1)/2;
        }
        return count;
    }
}